AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -3468,6 +3440,27 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return [self layoutThatFits:constrainedSize parentSize:constrainedSize.max];
 }
 
+- (void)setPreferredFrameSize:(CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // Deprecated preferredFrameSize just calls through to set width and height
+  self.width = ASDimensionMake(preferredFrameSize.width);
+  self.height = ASDimensionMake(preferredFrameSize.height);
+  [self invalidateCalculatedLayout];
+}
+
+- (CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  // We don't support fractions in preferredFrameSize
+  ASDisplayNodeAssert(self.width.unit != ASDimensionUnitFraction, @"preferredFrameSize.width cannot be of unit type ASDimensionUnitFraction");
+  ASDisplayNodeAssert(self.height.unit != ASDimensionUnitFraction, @"preferredFrameSize.height cannot be of unit type ASDimensionUnitFraction");
+
+  return CGSizeMake(self.width.value, self.height.value);
```

Comment:
When do you think we should return `CGSizeZero`? We could return zero if width and height are of type `auto`, but that's already happening. What if just one of `width` / `height` properties have a unit of `ASDimensionUnitFraction` and developer expect some size if at least one is of type `ASDimensionUnitPoint`?